File: 03.business-os/_archive/commonos_legacy_skeleton_20260425_184835/app_local/PocketSecretary/iphone/PocketSecretary/CommonOSLocalQueue/CommonOSLocalQueueRegistry.swift
import Foundation

enum CommonOSLocalQueueRegistry {
    static func createDefault() -> CommonOSLocalQueueDescriptor {
        CommonOSLocalQueueDescriptor(
            appName: "PocketSecretary",
            syncMode: "offline-first",
            syncTriggers: [
                "app_launch",
                "foreground_resume",
                "online_recovery",
                "manual_sync",
                "send_possible"
            ],
            retryStates: [
                "retry_wait",
                "failed",
                "processing"
            ],
            queueKinds: [
                "briefing_refresh",
                "follow_through_update",
                "conversation_action_write"
            ],
            structuredStateStorage: "indexeddb_equivalent",
            localOutboxStorage: "local_outbox_queue"
        )
    }
}

File: 03.business-os/PocketSecretary/android/app/src/main/java/com/lsam/PocketSecretary/runtime/commonos/localqueue/CommonOSLocalQueueRegistry.java
package com.lsam.PocketSecretary.runtime.commonos.localqueue;

import java.util.ArrayList;
import java.util.List;

public final class CommonOSLocalQueueRegistry {

    public CommonOSLocalQueueDescriptor createDefault() {
        final List<String> syncTriggers = new ArrayList<>();
        syncTriggers.add("app_launch");
        syncTriggers.add("foreground_resume");
        syncTriggers.add("online_recovery");
        syncTriggers.add("manual_sync");
        syncTriggers.add("send_possible");

        final List<String> retryStates = new ArrayList<>();
        retryStates.add("retry_wait");
        retryStates.add("failed");
        retryStates.add("processing");

        final List<String> queueKinds = new ArrayList<>();
        queueKinds.add("briefing_refresh");
        queueKinds.add("follow_through_update");
        queueKinds.add("conversation_action_write");

        return new CommonOSLocalQueueDescriptor(
                "PocketSecretary",
                "offline-first",
                syncTriggers,
                retryStates,
                queueKinds,
                "indexeddb_equivalent",
                "local_outbox_queue"
        );
    }
}
